File: 0605_can_place_flowers/solution.py
# LeetCode 0605 - Can Place Flowers
# https://leetcode.com/problems/can-place-flowers/

from typing import List


class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True

        bed = flowerbed[:]
        for i in range(len(bed)):
            if bed[i] == 1:
                continue
            left_empty = i == 0 or bed[i - 1] == 0
            right_empty = i == len(bed) - 1 or bed[i + 1] == 0
            if left_empty and right_empty:
                bed[i] = 1
                n -= 1
                if n == 0:
                    return True
        return False

File: 0605_can_place_flowers/solution.go
// LeetCode 0605 - Can Place Flowers
// https://leetcode.com/problems/can-place-flowers/

func canPlaceFlowers(flowerbed []int, n int) bool {
	if n == 0 {
		return true
	}
	bed := append([]int(nil), flowerbed...)
	for i := 0; i < len(bed); i++ {
		if bed[i] == 1 {
			continue
		}
		leftEmpty := i == 0 || bed[i-1] == 0
		rightEmpty := i == len(bed)-1 || bed[i+1] == 0
		if leftEmpty && rightEmpty {
			bed[i] = 1
			n--
			if n == 0 {
				return true
			}
		}
	}
	return false
}
